demos/common/cpp/utils/include/utils/ocv_common.hpp

```diff
@@ -85,9 +85,8 @@ static UNUSED void matToBlob(const cv::Mat& mat, const InferenceEngine::Blob::Pt
 * @param tensor - Tensor object which to be filled by an image data.
 * @param batchIndex - batch index of an image inside of the blob.
 */
-static UNUSED void matToTensor(const cv::Mat& mat, const ov::Tensor& tensor, int batchIndex = 0) {
+static UNUSED void matToTensor(const cv::Mat& mat, const ov::Tensor& tensor, const ov::Layout& layout = {"NCHW"}, int batchIndex = 0) {
     ov::Shape tensorShape = tensor.get_shape();
-    ov::Layout layout("NCHW");
     const size_t width = tensorShape[ov::layout::width_idx(layout)];
     const size_t height = tensorShape[ov::layout::height_idx(layout)];
     const size_t channels = tensorShape[ov::layout::channels_idx(layout)];
```

Comment:
These three lines are the only usage of `layout`. But if you really expect anything else but `NCHW`, you need to adopt loops which perform copy from mat to tensor.

My suggestions is to agree that `matToTensor()` may take only `NHWC`. Otherwise the conversion should have been done by `PrePostProcessor`. The agreement transforms your `matToTensor()` to my `resize2tensor()`: https://github.com/openvinotoolkit/open_model_zoo/pull/3103/files#diff-6abf992f3a0ccc34e1727aa9a219783e02b1ba3ec10b1227d86aa20ce346ed9cR185

Reply:
Also, it seems, `matToTensor()` isn't needed in this demo. `wrapMat2Tensor()` should be enough

Reply:
I'm pretty fine with removing functions which are not needed anymore but the case is you have OV2.0 app and user supply model in IR v10 from previous release (most obvious case for evaluating new release). We need to handle this somehow.

Reply:
Does `PrePostProcessor` work for IRv10?

Reply:
I've dismiss these changes